angularSTEP/src/app/discover-display/discover-display.component.ts

```diff
@@ -64,134 +65,66 @@ export class DiscoverDisplayComponent implements OnInit {
         }
-        break;
-      }
-      case 'Name': {
-        if (this.direction) {
-          this.recipeCollection.ref
-            .orderBy('recipeName', 'desc')
-            .withConverter(new RecipeConverter().recipeConverter)
-            .get()
-            .then(recipes => {
-              this.recipes = recipes.docs;
-            });
-        } else {
-          this.recipeCollection.ref
-            .orderBy('recipeName', 'asc')
-            .withConverter(new RecipeConverter().recipeConverter)
-            .get()
-            .then(recipes => {
-              this.recipes = recipes.docs;
-            });
+        case 'Name': {
+          this.getDocs('recipes', 'recipeName', this.direction ? 'desc' : 'asc');
+          break;
         }
-        break;
       }
     }
   }
 
   getUsers() {
     switch (this.sortType) {
       case 'Time Created': {
-        if (this.direction) {
-          this.userCollection.ref
-            .orderBy('time', 'asc')
-            .withConverter(new Converter().userConverter)
-            .get()
-            .then(users => {
-              this.users = users.docs;
-            });
-        } else {
-          this.userCollection.ref
-            .orderBy('time', 'desc')
-            .withConverter(new Converter().userConverter)
-            .get()
-            .then(users => {
-              this.users = users.docs;
-            });
-        }
+        this.getDocs('users', 'time', this.direction ? 'asc' : 'desc');
         break;
       }
-      case 'Number of Recipes': {
-        //temporary until num of recipes are actually added
-        if (this.direction) {
-          this.userCollection.ref
-            .orderBy('displayName', 'desc')
-            .withConverter(new Converter().userConverter)
-            .get()
-            .then(users => {
-              this.users = users.docs;
-            });
-        } else {
-          this.userCollection.ref
-            .orderBy('displayName', 'asc')
-            .withConverter(new Converter().userConverter).limit(10)
-            .get()
-            .then(users => {
-              this.users = users.docs;
-            });
-        }
+      case 'Name': {
+        this.getDocs('users', 'displayName', this.direction ? 'desc' : 'asc');
         break;
       }
-      case 'Name': {
-        if (this.direction) {
-          this.userCollection.ref
-            .orderBy('displayName', 'desc')
-            .withConverter(new Converter().userConverter).limit(10)
-            .get()
-            .then(users => {
-              this.users = users.docs;
-            });
-        } else {
-          this.userCollection.ref
-            .orderBy('displayName', 'asc')
-            .withConverter(new Converter().userConverter).limit(10)
-            .get()
-            .then(users => {
-              this.users = users.docs;
-            });
-        }
+    }
+  }
+
+  getDocs(type: string, sort: string, direction: firebase.firestore.OrderByDirection) {
```

Comment:
WDYT of splitting into two functions for getUsers() and getRecipes() since that's effectively what the "type" argument does here anyway (no shared code between the two cases) and it eliminates the possibility for passing in an unknown "type"